src/containers/Consulting/Consulting.tsx

```diff
@@ -0,0 +1,232 @@
+    staff,
+    when,
+    duration,
+    content,
+    isBillable,
+    organization,
+  };
+
+  const { data: organizationList } = useQuery(FILTER_ORGANIZATIONS, {
+    variables: setVariables(),
+  });
+
+  if (!organizationList) {
+    return <Loading />;
+  }
+
+  const setStates = ({
+    participants: supportMembers,
+    staff: staffMembers,
+    when: consultingDate,
+    duration: durationInMin,
+    content: description,
+    isBillable: billable,
+    organization: org,
+    organizationName,
+  }: any) => {
+    setParticipants(supportMembers);
+    setStaff(staffMembers);
+    setWhen(consultingDate);
+    setDuration(durationInMin);
+    setContent(description);
+    setIsBillable(billable);
+    /**
+     * When edit organization is undefined
+     * since we are getting organizationName,
+     * we need to get object from organizationList for selected organizationName
+     */
+    if (!org) {
+      const selectedOrg = organizationList.organizations.find(
+        ({ name }: { name: string }) => name === organizationName
+      );
+      setOrganization(selectedOrg);
+    } else setOrganization(org);
+  };
+
+  const FormSchema = Yup.object().shape({
+    participants: Yup.string().required(t('Participants are required.')),
+    staff: Yup.string().required(t('NGO staff members are required.')),
+    content: Yup.string().required(t('Description is required')),
+    when: Yup.date().nullable().required(t('Date is required and should be valid.')),
+    duration: Yup.number()
+      .nullable()
+      .moreThan(0, 'Duration should be greater than 0')
+      .required(t('Duration is required')),
+    isBillable: Yup.boolean().nullable().required('Required'),
+    organization: Yup.object().nullable().required('Organization is required'),
+  });
+
+  const queries: any = {
+    getItemQuery: GET_CONSULTING_HOURS_BY_ID,
+    createItemQuery: CREATE_CONSULTING_HOUR,
+    updateItemQuery: UPDATE_CONSULTING_HOURS,
+    deleteItemQuery: DELETE_CONSULTING_HOURS,
+  };
+
+  const formFields = (organizationOptions: Array<any>) => [
+    {
+      component: AutoComplete,
+      name: 'organization',
+      placeholder: t('Select Organization'),
+      options: organizationOptions,
+      optionLabel: 'name',
+      multiple: false,
+      textFieldProps: {
+        label: t('Select Organization'),
+        variant: 'outlined',
+      },
+      onChange: (val: any) => setOrganization(val),
+    },
+    {
+      component: Input,
+      name: 'participants',
+      type: 'text',
+      placeholder: t('Participants'),
+      inputProp: {
+        onChange: (event: any) => setParticipants(event.target.value),
+      },
+    },
+    {
+      component: DateTimePicker,
+      name: 'when',
+      placeholder: t('Select date'),
+      onChange: (val: any) => setWhen(val),
+    },
+    {
+      component: Input,
+      name: 'duration',
+      type: 'text',
+      placeholder: t('Enter time (in mins)'),
+      validate: (val: any) => {
```

Comment:
Validation should be done through formik.

Reply:
This was custom validation so I've added validate function. But we can replace this using `Yup.number().test() `

Reply:
Test functions got 3 params
- test name
- validation error message
- validation function which returns true/false

![image](https://user-images.githubusercontent.com/26791171/118441103-4f1e3a80-b706-11eb-90e0-866d38b1c6d5.png)